.github/workflows/workflow-build-push-container-github-registry.yml

```diff
@@ -0,0 +1,124 @@
+            echo "::set-output name=exists::true"
+          else
+            echo "::set-output name=exists::false"
+          fi
+
+      - name: Cancel the workflow if no Dockerfile is present
+        if: steps.check-file.outputs.exists == 'false'
+        uses: andymckay/cancel-action@0.2
+
+      # needed since the run is not cancelled immediately
+      - name: Wait for run cancellation
+        if: steps.check-file.outputs.exists == 'false'
+        shell: bash
+        run: |
+          echo "No Dockerfile found, cancelling the workflow."
+          while true; do echo "Waiting for job to be cancelled"; sleep 5; done
+
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Log in to the github container registry (GCR)
+        uses: docker/login-action@65b78e6e13532edd9afa3aa52ac7964289d1a9c1
+        with:
+          registry: ${{ inputs.registry }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Cache Docker layers
+        uses: actions/cache@v3
+        with:
+          path: /tmp/.buildx-cache
+          key: ${{ runner.os }}-buildx-${{ github.sha }}
+          restore-keys: ${{ runner.os }}-buildx
+
+      - name: Extract branch name
+        shell: bash
+        run: echo "branch=${GITHUB_HEAD_REF:-${GITHUB_REF#refs/heads/}}" >> $GITHUB_OUTPUT
+        id: extract_branch_name
+
+      - name: Build and push (3 tags | commit number, pr number, branch name)
+        id: docker_build_and_tag
+        uses: docker/build-push-action@v5
+        with:
+          context: .
+          push: true
+          tags: |
+            ${{ inputs.registry }}/${{ inputs.container-name }}:${{ github.event.number }}
+            ${{ inputs.registry }}/${{ inputs.container-name }}:${{ inputs.tag }}
+            ${{ inputs.registry }}/${{ inputs.container-name }}:${{ steps.extract_branch_name.outputs.branch }}
+          cache-from: type=local,src=/tmp/.buildx-cache
+          cache-to: type=local,mode=max,dest=/tmp/.buildx-cache-new
+
+      - name: Refresh Cache
+        run: |
+          rm -rf /tmp/.buildx-cache
+          mv /tmp/.buildx-cache-new /tmp/.buildx-cache
+
+      - name: Image digest (latest)
+        run: echo ${{ steps.docker_build_and_tag.outputs.digest }}
+
+  remove-old-image:
+    runs-on: ubuntu-latest
+    needs: build-push-image
+    steps:
+      - name: Check Out Repo
+        uses: actions/checkout@v4
+
+      - name: Set up Docker
+        uses: docker/setup-buildx-action@v3
+
+      - name: Set up Python
+        uses: actions/setup-python@v3
+        with:
+          python-version: 3.8
+
+      - name: Install Python dependencies
+        run: pip install requests
+
+      - name: Download the python script from https://github.com/ai-cfia/github-workflows
+        run: |
+          curl -LO https://raw.githubusercontent.com/ai-cfia/github-workflows/main/scripts/remove-old-image.py
```

Comment:
hacky. make a reusable workflow in its own repo

Reply:
So should I make a repository for scripts?

Reply:
yes

Reply:
It's done. You can check in this pull request (https://github.com/ai-cfia/devops/pull/27).